### PR TITLE
vere: adds leer-jet in support of new +to-wain:format

### PR DESCRIPTION
**Pull request by joemfb**
This PR includes on the vere changes from #3180, which has the context and previous discussion.

### CODE REVIEW

File: pkg/urbit/jets/e/lore.c [hunk 1]
```diff
@@ -3,6 +3,30 @@
...
+    }
+    else {
+      // as an optimization, we alloc small lines on the stack
+      // (only small to prevent stack overflow)
+      c3_t  big_t = len_w > 255;
+      c3_y* buf_y = big_t ? u3a_malloc(len_w) : alloca(len_w);
+      u3_noun red;
+
+      u3r_bytes(pos_w, len_w, buf_y, src);
+      red = u3i_bytes(len_w, buf_y);
```
Comment (joemfb) [inline pkg/urbit/jets/e/lore.c:20]: Once #3612 is merged, this should be updated to use the new slab api and only allocate once.

**Comment by joemfb** [comment]
This PR is also a good use-case for the optimization in #3658.

**Comment by joemfb** [comment]
@philipcmonk, bump for review.

**Comment by joemfb** [comment]
I've updated this jet to use `u3i_defcons()` (ie, construct the output list in-order), refactored it slightly, and renamed the file to `leer.c` to better facilitate future maintenance/removal of %lore.

/cc @frodwith

**Review by Fang- — APPROVED** [review]
Confirmed sane!

**Pull request merged**